### PR TITLE
Updated public license FAQ [fixes #14137]

### LINKED ISSUE #14137: Update to the Mozilla Public License FAQ
Hi there. We have received a request to do update [THIS](https://www.mozilla.org/en-US/MPL/2.0/FAQ/) page. 

The items that need to be updated are in red font as tracked changes in [THIS](https://docs.google.com/document/d/1P2w5KGg-DAc54zP-BW1l0EqqKM50jSXXd1Ez9JXAzvo/edit#heading=h.iu86a4wxyqi5) document. 

**There are three changes to the page:**
1. Add “last updated” information toward the top
2. An addition to the answer to Question 5
3. A new Question 27 at the bottom

No l10n required. Completion within a couple of weeks is fine.

Please let me know if you have any questions. Thank you!!

### PR DESCRIPTION
**Pull request by reemhamz**
## One-line summary
Some updates came in for the Public License FAQ page.

[Copy doc](https://docs.google.com/document/d/1P2w5KGg-DAc54zP-BW1l0EqqKM50jSXXd1Ez9JXAzvo/edit)

## Issue / Bugzilla link
https://github.com/mozilla/bedrock/issues/14137

## Testing
http://localhost:8000/en-US/MPL/2.0/FAQ/

### CODE REVIEW

File: bedrock/mozorg/templates/mozorg/mpl/2.0/faq.html [hunk 1]
```diff
@@ -19,6 +23,9 @@ <h2 id="about-this-faq">About This FAQ</h2>
     <p>Please note that, while this FAQ is intended to be accurate and helpful, it <em>is not the license</em>, and may not cover important issues that affect you and your specific situation. As a result, reading the FAQ should not serve as a substitute for reading the license itself, or for seeking legal advice from a lawyer.</p>
     <p>If you see any errors in this FAQ, or have suggestions for further questions, please email <a href="mailto:licensing@mozilla.org">licensing@mozilla.org</a>.</p>
     <p>The MPL version 1.1 FAQ is still available <a href="{{ url('mozorg.mpl.1.1.faq') }}">here</a> but should not be used except as a reference for that license.</p>
+
+    <p class="last-updated">Last updated on January 30th, 2024
```
Comment (reemhamz) [inline bedrock/mozorg/templates/mozorg/mpl/2.0/faq.html:27]: Is there a way to automate this without having to refer to a Python file like what we do in `credits.html`?

File: bedrock/mozorg/templates/mozorg/mpl/2.0/faq.html [hunk 2]
```diff
@@ -6,6 +6,10 @@
 
 {% extends "mozorg/base.html" %}
 
+{% block page_css %}
+    {{ css_bundle('license-faq') }}
+  {% endblock %}
```
Comment (alexgibson) [inline bedrock/mozorg/templates/mozorg/mpl/2.0/faq.html:11]: Nit: fix indentation

File: bedrock/mozorg/templates/mozorg/mpl/2.0/faq.html (hunk 1 above)
Comment (alexgibson) [reply bedrock/mozorg/templates/mozorg/mpl/2.0/faq.html:27]: Not as far as I'm aware, at least without adding more work. This page rarely gets updated, so perhaps doing things manually is OK?

**Review by alexgibson — APPROVED** [review]
r+wc

**Pull request merged**